glTF-BinExporter/RhinoDocGltfConverter.cs

```diff
@@ -176,18 +183,36 @@ int[] GetMaterials(RenderMaterial[] materials, RhinoObject rhinoObject)
                     return null;
                 }
 
+                Guid materialId;
                 if (material == null && options.UseDisplayColorForUnsetMaterials)
                 {
-                    Color4f objectColor = GetObjectColor(rhinoObject);
-                    materialIndices[i] = CreateSolidColorMaterial(objectColor, GetObjectName(rhinoObject));
+                    if (options.ExportLayers)
+                    {
+                        if (subObjects[i].Attributes.ColorSource == ObjectColorSource.ColorFromLayer)
+                        {
+                            materialIndices[i] = GetLayerMaterial(rhinoObject);
+                            continue;
+                        }
+                    }
+
+                    Color4f objectColor = GetObjectColor(subObjects[i]);
+                    if (!colorIndices.TryGetValue(objectColor, out int colorIndex))
+                    {
+                        colorIndex = CreateSolidColorMaterial(objectColor, GetObjectName(rhinoObject));
+                        colorIndices.Add(objectColor, colorIndex);
+                    }
+                    materialIndices[i] = colorIndex;
                     continue;
                 }
                 else if (material == null)
                 {
                     material = Rhino.DocObjects.Material.DefaultMaterial.RenderMaterial;
+                    materialId = new Guid();
```

Comment:
Should probably use the [Guid.Empty](https://docs.microsoft.com/en-us/dotnet/api/system.guid.empty?view=net-5.0) here for the zero guid.